docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,583 @@
+
+If you do not have identical types, the best solution is to use the special
+`to_mathint` operator to convert both sides to `mathint`.  For example:
+
+```cvl
+assert to_mathint(balanceOf(user)) == initial + deposit;
+```
+
+Note that in this example, we do not need to cast the right hand side, since
+the result of `+` is always of type `mathint`.
+
+````{todo}
+When should you not simply cast to `mathint`?  We have one example: consider the
+following code:
+
+```cvl
+ghost uint256 sum;
+
+hook ... {
+    havoc sum assuming sum@new == sum@old + newBalance - oldBalance;
+}
+```
+
+Simply casting to `mathint` will turn overflows into vacuity.  It's not clear
+how to generalize from this example.
+
+In this particular example, the right solution is to declare `sum` to be a
+`mathint` instead of a `uint`.  Note that with the more "modern" update syntax,
+this isn't a problem:
+
+```cvl
+ghost uint256 sum;
+
+hook ... {
+    sum = sum + newBalance - oldBalance;
+}
+```
+will say that the right-hand side is a `mathint` which can't be assigned to a
+`uint`.
+
+We hope that security engineers will think carefully about these changes and let
+us know of any other situations where the right thing is not simply casting to
+`mathint`.
+````
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+(cvl2-casting)=
+### Implicit and explicit casting
+
+If every number that can be represented by one type can also be represented by
+another type, then we say that the first type is a *subtype* of the second type.
+
+For example, a `uint8` variable could have any value between `0` and `2^8-1`,
+and all of these values can be stored in a `uint16` variable, so `uint8` is a
+subtype of `uint16`.  An `int16` can also store any value between `0` and
+`2^8-1`, so `uint8` is also a subtype of `int16`.
+
+All integer types are subtypes of `mathint`, since any integer can be
+represented by a `mathint`.
+
+In CVL 1, the rules for converting between supertypes and subtypes were
+complicated; they depended not only on the types involved, but on the context
+in which the conversion happened.  CVL 2 simplifies these rules and improves the
+clarity and predictability of casts.
+
+In CVL 2, with one exception, you can always use a subtype whenever the
+supertype is accepted.  For example, you can always use a `uint8` where an
+`int16` is expected.  We say that the subtype can be "implicitly cast" to the
+supertype.
+
+The one exception is comparison operators; as mentioned above, you must add an
+explicit conversion if you want to compare two numbers with different types.
+The `to_mathint` operator exists solely for this purpose; in all other contexts
+you can simply use any number when a `mathint` is expected (since all integer
+types are subtypes of `mathint`).
+
+In order to convert from a supertype to a subtype, you must use an explicit
+cast.  In CVL 1, the syntax for casting to a subtype was `to_<subtype>(value)`,
+for example `to_uint256(x)`.
```

Comment:
We should mention those are no longer available in CVL2

Reply:
and I think **only** `to_uint256` existed, other versions didn't